### PR TITLE
Fixing Flavors & App Bundle missing index.android.bundle with proguard enabled 

### PR DESCRIPTION
**Pull request by renopp**
## Summary

This commit fixes Flavors & App Bundle missing index.android.bundle with proguard enabled 
fixes https://github.com/facebook/react-native/issues/31157

## Changelog

[Android] [Fixed] - Fixing Flavors & App Bundle missing index.android.bundle with proguard enabled 

## Test Plan

Manual
run `./gradlew clean` then `./gradlew bundleRelease `

### CODE REVIEW

**Comment by renopp** [comment]
@gedeagas

**Review by gedeagas — APPROVED** [review]
LGTM

**Comment by davwheat** [comment]
For reference, the failed test is unrelated to this PR.

**Comment by gedeagas** [comment]
@renopp head's up, your PR are being reviewed internally at FB.

**Comment by cortinico** [comment]
Thanks for your PR @renopp 
Can I ask you to rebase on top of `main`. The CI run is quite old and it should be green by now 👍

**Comment by renopp** [comment]
Hi, @cortinico Thanks for the review

I have rebase the PR and then the CI run has been green by now ✅

File: react.gradle [hunk 1]
```diff
@@ -315,6 +315,7 @@ afterEvaluate {
         // mergeResources task runs before the bundle file is copied to the intermediate asset directory from Android plugin 4.1+.
         // This ensures to copy the bundle file before mergeResources task starts
         def mergeResourcesTask = tasks.findByName("merge${targetName}Resources")
+        currentAssetsCopyTask.dependsOn(currentBundleTask)
```
Comment (cortinico) [inline react.gradle:318]: Isn't this line equivalent to this other?

https://github.com/facebook/react-native/blob/2e8893fb5be0c877e0818d04264d75fa92ae0f8e/react.gradle#L312

**Comment by cortinico** [comment]
As a side note, we merged this last week #32026 that is introducing the line I mentioned in the comment.

So I believe the issue is now fixed and we don't need to merge this PR anymore.
Can I ask you to verify that the issue is now solved on your end?

File: react.gradle (hunk 1 above)
Comment (renopp) [reply react.gradle:318]: Yeah, I think that's equivalent.

I have tried to replicate the issue at RN 65.1 then the issue doesn't appear.
I will close this PR.

Thanks 🍻

**Pull request closed without merging**